### PR TITLE
User updates

### PR DESCRIPTION
**Pull request by jonnykry**
# Overview

This PR implements authentication and exposing UserID of the current authenticated user to the rest of the application.

### CODE REVIEW

File: src/components/Register.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ class Register extends Component {
                         </select>
                         <div className="mt4"><input className="w-25 b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6" type="submit" value="Submit" required /></div>
                         {this.props.error && (
-                            <p>Bad login information</p>
+                            <p>Something didn't work out, please try again.</p>
```
Comment (ghmeier) [inline src/components/Register.js]: Just as an FYI, I created a `SuccessMessage` and `ErrorMessage` component to handle this common case.

File: src/components/Register.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ class Register extends Component {
...
                         )}
                     </form>
                 </article>
```
Comment (ghmeier) [inline src/components/Register.js]: Also, you need to add a `mapStateToProps` function to make sure that state is passed into this file if we want it.

File: src/components/RegisterContainer.js [hunk 3]
```diff
@@ -36,17 +43,14 @@ class RegisterContainer extends Component {
 
             dispatch(createUser(data));
```
Comment (ghmeier) [inline src/components/RegisterContainer.js]: We should `mapStateToProps` in this component as well, correct?

**Review by ghmeier — APPROVED** [review]

File: src/components/Register.js (hunk 1 above)
Comment (jonnykry) [reply src/components/Register.js]: Right on, I'll hook into those!

File: src/components/Register.js (hunk 2 above)
Comment (jonnykry) [reply src/components/Register.js]: Yup!  Still waiting on the backend changes, then I'll verify passing through the UserID works.

Comment (jakelong95) [reply src/components/Register.js]: In case you hadn't seen, the TownCenter update is live now

**Pull request merged**